app/Components/Matrix.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Button from './commonComponent/Button.jsx';
+
+
+class Matrix extends React.Component {
+    static propTypes = {
+        rows: PropTypes.number,
+        columns: PropTypes.number
+    
+    };
+    static defaultProps = {
+    rows: 16,
+    columns: 16
+  };
+  constructor(props) {
+      super(props);
+      this.positionValues = ['TOP','RIGHT','BOTTOM','LEFT'];
+      this.matrix= this.generateTable(this.props.rows, this.props.columns);
+      let currentPosition = this.matrix[0];
+      this.state = {
+          currentX: currentPosition.x,
+            currentY: currentPosition.y,
+          currentPosition: this.positionValues[1],
```

Comment:
there are 
let currentPosition = this.matrix[0];
and 
currentPosition: this.positionValues[1]

these two are confusing. they should not share the same name